rio_tiler_pds/sentinel/utils.py

```diff
@@ -48,14 +48,14 @@ def s2_sceneid_parser(sceneid: str) -> Dict:
         )
 
     elif re.match(
-        "^S2[AB]_[0-9]{2}[A-Z]{3}_[0-9]{8}_[0-9]_L[0-2][A-C]$", sceneid
+        "^S2[AB]_[0-9]{1,2}[A-Z]{3}_[0-9]{8}_[0-9]_L[0-2][A-C]$", sceneid
```

Comment:
I guess it's ok to change the old style scene id too. Can you do it @kylebarron. would also be nice to add entry in changelog